code/modules/power/lighting.dm

```diff
@@ -514,14 +514,12 @@
 	status = LIGHT_EMPTY
 	update()
 
-/obj/machinery/light/proc/broken(skip_sound_and_sparks = 0)
-	if(status == LIGHT_EMPTY || status == LIGHT_BROKEN)
+/obj/machinery/light/proc/broken(skip_sounds_and_sparks = 0)
+	if(status == LIGHT_EMPTY || status == LIGHT_BROKEN || status == LIGHT_BURNED)
 		return
-
-	if(!skip_sound_and_sparks)
-		if(status == LIGHT_OK || status == LIGHT_BURNED)
+	if(!skip_sounds_and_sparks)
+		if(status == LIGHT_OK)
 			playsound(src.loc, 'sound/effects/Glasshit.ogg', 75, 1)
-		if(on)

```

Comment:
Here's where @Krausus wanted you to make a change.

something like

`if(on || always_sparks)`

then you'd add an `always_sparks` argument to `broken` with it defaulting to `0` but specifically set to `1` for the `overload_lighting` proc.